### PR TITLE
Rewrite `TestOutputTables`

### LINKED ISSUE #20: `groupby-2-way` tables are not written to the correct directory
Given the following table configuration (the `tables` property)...

```yaml
tables:
  two_way:
    tab_type: 2-way
    variables:
      - sex
      - age_band
  target_two_way:
    tab_type: target-2-way
    target: has_copd
    variables:
      - sex
      - age_band
  groupby_two_way:
    tab_type: groupby-2-way
    groupby: age_band
    variables:
      - sex
      - has_copd
    outputs:
      moderately_sensitive:
        safetab_log: output/input_tables/table_log.txt
        safetab_two_way: output/input_tables/two_way/*.md
        safetab_target_two_way: output/input_tables/target_two_way/*.md
        safetab_groupby_two_way: output/input_tables/groupby_two_way/*.md
```

...the `groupby-2-way` table called `groupby_two_way` is written to the `target_two_way` directory.

### PR DESCRIPTION
**Pull request by iaindillingham**
TL;DR There was a bug in the test, not in the function under test. The test has been rewritten, breaking it up into three tests and removing patched function calls.

Previously, this test class had a single test method that heavily relied on patching calls made within the `output_tables` function (the function under test). An incorrect return value from a patched call suggested that there was a bug; namely, that tables of type `groupby-2-way` were not being written to the correct directory. By splitting this test class into three test methods, one for each type of table, and by writing an input CSV to a temporary directory rather than by patching, we can confirm that there isn't a bug. As a bonus, we get a better test class, too.

Closes #20

### CODE REVIEW

**Review by inglesp — APPROVED** [review]
A very obvious improvement, and much more readable.  (The rest of the codebase looks very readable too, nice work.)

I wonder whether the tests could assert something stronger.  At the moment, we're just asserting that `output_table()` creates some files.  But looking at the code in `create_tables.py`, there's a lot going on.  Should we be asserting that the contents of the files are as expected?  As well as giving us assurance that the code is doing the right thing, such tests might also be useful for documentation.  (Probably something for the backburner.)

**Comment by iaindillingham** [comment]
I'm less keen on the rest of the codebase: there are some very long functions, even longer when we consider that they are essentially chained calls 🙂

I'd like to make more assertions about the contents of the files, but in this case, that would probably mean grepping the files*. I think that I'd prefer to follow a test-refactor-test cycle. This PR is the first test; next I was planning to move the loops in `output_tables` into separate functions; then I was planning to test these functions.

*Not quite true! We could count the number of lines in the log file and also check that when a table has been redacted, its corresponding directory has been created. The former seems like a nice-to-have; the latter seems like an implementation accident.
